packages/features/ee/managed-event-types/reassignment/services/ManagedEventReassignmentService.ts

```diff
@@ -0,0 +1,252 @@
+
+  private async findEligibleReassignmentUsers(
+    parentId: number,
+    currentUserId: number | null,
+    orgId: number | null,
+    log: typeof logger
+  ): Promise<IsFixedAwareUser[]> {
+    const allChildEventTypes = await this.eventTypeRepository.findManyChildEventTypes(
+      parentId,
+      currentUserId
+    );
+
+    const userIds = allChildEventTypes.map((et) => et.userId).filter((id): id is number => id !== null);
+
+    if (userIds.length === 0) {
+      throw new Error("No other users available for reassignment in this managed event");
+    }
+
+    const usersWithSelectedCalendars =
+      await this.userRepository.findManyByIdsWithCredentialsAndSelectedCalendars({
+        userIds,
+      });
+
+    log.info(`Found ${usersWithSelectedCalendars.length} potential reassignment targets`);
+
+    const enrichedUsers = await enrichUsersWithDelegationCredentials({
+      orgId,
+      users: usersWithSelectedCalendars,
+    });
+
+    const usersWithSchedules = enrichedUsers.filter((user) => {
+      const hasSchedules = user.schedules && user.schedules.length > 0;
+      if (!hasSchedules) {
+        log.warn(`User ${user.id} skipped: no schedules configured`);
+      }
+      return hasSchedules;
+    });
+
+    if (usersWithSchedules.length === 0) {
+      throw new Error(
+        "No eligible users found for reassignment. All team members must have availability schedules configured."
+      );
+    }
+
+    log.info(`${usersWithSchedules.length} users with schedules configured`);
+
+    return usersWithSchedules.map((user) => ({
+      ...user,
+      isFixed: false as const,
+    })) as unknown as IsFixedAwareUser[];
+  }
+
+  private async filterUsersAvailableAtTime(
+    users: IsFixedAwareUser[],
+    parentEventType: NonNullable<Awaited<ReturnType<typeof getEventTypesFromDB>>>,
+    timeSlot: { start: Date; end: Date; timeZone: string },
+    log: typeof logger
+  ): Promise<IsFixedAwareUser[]> {
+    const availableUsers = await ensureAvailableUsers(
+      { ...parentEventType, users },
+      {
+        dateFrom: dayjs(timeSlot.start).format(),
+        dateTo: dayjs(timeSlot.end).format(),
+        timeZone: timeSlot.timeZone,
+      },
+      logger
+    );
+
+    if (availableUsers.length === 0) {
+      throw new Error("No users available at the booking time");
+    }
+
+    log.info(`${availableUsers.length} users available at booking time`);
+
+    return availableUsers;
+  }
+
+  private async selectReassignmentUser(
+    availableUsers: IsFixedAwareUser[],
+    parentEventType: NonNullable<Awaited<ReturnType<typeof getEventTypesFromDB>>>,
+    log: typeof logger
+  ): Promise<IsFixedAwareUser> {
+    if (availableUsers.length === 0) {
+      throw new Error("No available users to select for reassignment");
+    }
+
+    const nonEmptyUsers: [IsFixedAwareUser, ...IsFixedAwareUser[]] =
+      availableUsers.length === 1
+        ? [availableUsers[0]]
+        : [availableUsers[0], ...availableUsers.slice(1)];
```

Comment:
shouldn't this be enough `const nonEmptyUsers = availableUsers as [IsFixedAwareUser, ...IsFixedAwareUser[]];`?